File: src/main/java/org/gangel/graphomance/usecases/CreateSingleVertexCompoundUniqueHashIndex.java
package org.gangel.graphomance.usecases;

import org.gangel.graphomance.DbType;
import org.gangel.graphomance.IndexType;

public class CreateSingleVertexCompoundUniqueHashIndex extends CreateSingleVertexBase {

    public CreateSingleVertexCompoundUniqueHashIndex() {
        super("Create vertex with compound unique hash index", IndexType.HASH_UNIQUE, true, true);
    }

    @Override
    public boolean skipFor(DbType dbType) {
        return (dbType == DbType.NEO4J);
    }
}

File: src/main/kotlin/org/gangel/graphomance/usecases/node/CreateSingleVertexCompoundUniqueHashIndex.kt
package org.gangel.graphomance.usecases.node

import org.gangel.graphomance.api.DbType
import org.gangel.graphomance.api.IndexType

class CreateSingleVertexCompoundUniqueHashIndex :
	CreateSingleVertexBase("Create node with compound unique hash index",
						   IndexType.HASH_UNIQUE,
						   true,
						   true) {
	override fun skipFor(dbType: DbType): Boolean {
		return dbType == DbType.NEO4J
	}
}